syntactic/src/main/scala/syntactic_checker/whitelist/Features.scala

```diff
@@ -0,0 +1,319 @@
+    case _: Defn.Var => true
+    case _: Term.Assign => true
+    case _: Term.Do => true
+    case _: Term.Return => true
+    case _: Term.Throw => true
+    case _: Term.Try => true
+    case _: Term.TryWithHandler => true
+    case _: Term.While => true
+    case _: Pat.Var => true
+  })
+
+  /**
+   * Allows the use of contextual constructs, including:
+   * implicits, using, givens
+   */
+  case object ContextualConstructs extends AtomicFeature({
+    case Mod.Implicit() => true
+    case Mod.Using() => true
+    case _: Decl.Given => true
+    case _: Defn.Given => true
+    case _: Defn.GivenAlias => true
+    case _: Importee.Given => true
+    case _: Importee.GivenAll => true
+    case _: Pat.Given => true
+    case _: Type.ImplicitFunction => true
+    case _: Term.ApplyUsing => true
+  })
+
+  /**
+   * Allows the definition of extension groups and extension methods
+   */
+  case object Extensions extends AtomicFeature({
+    case _: Defn.ExtensionGroup => true
+  })
+
+  /**
+   * Allows the use of macro and metaprogramming-related constructs
+   */
+  case object Metaprogramming extends AtomicFeature({
+    case _: Defn.Macro => true
+    case _: Term.QuotedMacroExpr => true
+    case _: Term.QuotedMacroType => true
+    case _: Term.SplicedMacroExpr => true
+    case _: Term.SplicedMacroPat => true
+    case _: Type.Macro => true
+    case _: Pat.Macro => true
+    case _: internal.trees.Quasi => true
+  })
+
+  /**
+   * Allows the use of packages
+   */
+  case object Packages extends AtomicFeature({
+    case _: Pkg => true
+    case _: Pkg.Object => true
+  })
+
+  /**
+   * Allows the use of imports
+   */
+  case object Imports extends AtomicFeature({
+    case _: Import => true
+    case _: Importee => true
+    case _: Importer => true
+    case _: Name.Indeterminate => true
+  })
+
+  /**
+   * Allows export clauses
+   */
+  case object Exports extends AtomicFeature({
+    case _: Export => true
+    case _: Importee => true
+    case _: Importer => true
+    case _: Name.Indeterminate => true
+  })
+
+  /**
+   * Allows the use of XML-related constructs
+   */
+  case object Xml extends AtomicFeature({
+    case _: Term.Xml => true
+    case _: Pat.Xml => true
+    case _: Lit.String => true
+  })
+
+  /**
+   * Allows string interpolation
+   */
+  case object StringInterpolation extends AtomicFeature({
+    case _: Term.Interpolate => true
+    case _: Pat.Interpolate => true
+  })
+
+  /**
+   * Allows annotations
+   */
+  case object Annotations extends AtomicFeature({
+    case _: Term.Annotate => true
+    case _: Mod.Annot => true
+  })
+
+  /**
+   * Allows to define infix methods
+   */
+  case object Infixes extends AtomicFeature({
+    case Mod.Infix() => true
+  })
+
+  /**
+   * Allows to define inline methods
+   */
+  case object Inlines extends AtomicFeature({
+    case Mod.Inline() => true
+  })
+
+  /*
+  Not implemented  TODO figure out what they are
+
+  MultiSource
+  Term.ApplyType
```

Comment:
I found it: https://astexplorer.net/#/gist/ec56167ffafb20cbd8d68f24a37043a9/38a16f79d9dfbcb1b9247dcc7367216f10555424

It should be allowed in `PolymorphicTypes`.

Reply:
Done, thanks!